crates/pairwise_nostd_internal/src/accumulator.rs

```diff
@@ -30,86 +32,187 @@
+/// `glam` package or
+/// [`std::f64x2`](https://doc.rust-lang.org/std/simd/type.f64x2.html) from
+/// nightly. For this to work, we would probably to transmute the input
+/// array buffers. If using `glam`'s types, we might want to look into
+/// `bytemuck` to safely transmute. If using nightly, `std::simd` [has some
+/// support for doing this](https://doc.rust-lang.org/std/simd/struct.Simd.html#layout-1).
+///
+/// # General Optimization Notes
+/// We probably want to ensure that the accumulator methods are inlined. If
+/// they aren't inlined, then its conceivable that using a struct may
+/// introduce a performance penalty, especially if we start encoding
+/// (mathematical) vector components. In more detail, passing the values in
+/// a struct to a function, means that the values need to be packed in memory
+/// to match the struct's internal representation to communicate across the
+/// function boundaries (this issue goes away when the function is inlined).
+///
+/// There's definitely more to this story. As I understand it,
+/// platform-specific calling conventions will influence how values are passed
+/// to functions even if we passed each struct-member as individual args. Thus,
+/// we probably want to ensure inlining to get optimal code either way (i.e.
+/// there's no real disadvantage to using the struct, *yet*).
+///
+/// In practice, my real concern is in the context of SIMD optimizations.
+/// - I have some concerns that the packing multiple SIMD-sized data packs into
+///   a struct to pass the information into an inlined function, may present
+///   more of a challenge to the optimizer than just passing each SIMD-sized
+///   data pack directly to the function as arguments
+/// - First, while an optimizer should be able to make the same optimizations
+///   in both cases, the act of packing values into a struct certainly makes
+///   more work for the optimizer. It's unclear whether the "quantity of work"
+///   actually matters... (it seems plausible that an optimizer might have a
+///   heuristic telling it "give up" if there seems to be "too much" to avoid
+///   needlessly increasing compile times)
+/// - Second, I *think* the types offered by `std::simd` (e.g. `f64x2`) may
+///   receive special treatment from the rust compiler to encourage
+///   optimizations with SIMD operations (I may be wrong). If this is indeed
+///   the case, I have some concerns that placing such types within a struct
+///   *could* interfere with this treatment (due to the struct's memory
+///   representation requirements).
+///
+/// With all that said, I think it's worthwhile use an explicit type for now.
+/// I think that there are significant advantages to improving the readability
+/// of the code and it would be relatively move away from using the struct
+/// representation later (it would be easier to do that before we start
+/// introducing more accumulators).
+pub struct DataElement {
+    pub value: f64,
+    pub weight: f64,
+}
```

Comment:
But I think _maybe_ this could have a better name?  Ideas:
- `Datum`
- `DataPoint`
- `ConsumableValue`

Idk, it's also OK as is.

Reply:
Done. I like `Datum`. In fact I originally thought about naming it `Datum`, but I thought *maybe* `DataElement` was slightly more clear... But the fact that you like `Datum` is good enough for me!

Plus, I really like that it's short (I've typed it out a lot in PR #31)